modules/core/src/core/lib/pick-layers.js

```diff
@@ -119,13 +119,13 @@ export function pickObject(
       pixelRatio
     });
 
-    if (pPickInfos) {
-      pPickInfos.forEach(info => result.push(info));
+    if (processedPickInfos) {
+      processedPickInfos.forEach(info => result.push(info));
     }
   }
 
   // reset only affected buffers
-  Object.keys(exclude).forEach(layerId => layers[layerId].updateInstancePickingColors([]));
+  Object.keys(exclude).forEach(layerId => layers[layerId].updatePickingColors([]));
```

Comment:
Not sure I understand how this can restore the colors, in the non-instanced case. 

Unless we save ranges of cleared colors, we either:
* need a copy of the initial picking buffer
* or maybe just trigger the attributeManager to update the colors `attributeManager.invalidate('pickingColors')`.

Reply:
Working on it.